index.d.ts

```diff
@@ -0,0 +1,170 @@
+declare module 'fluture' {
+  interface ResolveFunction<R> {
+    (value: R): void
+  }
+  interface RejectFunction<L> {
+    (error: L): void
+  }
+
+  interface Cancel {
+    (): void
+  }
+
+  export abstract class IFuture<L, R> {
+    fork(reject: RejectFunction<L>, resolve: ResolveFunction<R>): Cancel
+  }
+
+  export function after<L, R>(duration: number, value: R): IFuture<L, R>
+
+  export function and<L, R>(
+    left: IFuture<L, any>,
+    right: IFuture<L, R>
+  ): IFuture<L, R>
+
+  export function ap<L, R, VR>(
+    apply: IFuture<L, (value: VR) => R>,
```

Comment:
Ah got it, thanks. So `Future.ap` is not specialised for the Future type (and it leverages `Z.ap` I guess)